wayland-scanner/src/c_interfaces.rs

```diff
@@ -27,14 +26,13 @@ pub(crate) fn generate_interfaces_prefix(protocol: &Protocol) -> TokenStream {
 
     let types_null_len = Literal::usize_unsuffixed(longest_nulls);
 
-    let nulls = repeat(quote! { NULLPTR as *const wayland_backend::protocol::wl_interface })
-        .take(longest_nulls);
-
     quote! {
-        const NULLPTR: *const std::os::raw::c_void = 0 as *const std::os::raw::c_void;
-        static mut types_null: [*const wayland_backend::protocol::wl_interface; #types_null_len] = [
-            #(#nulls,)*
-        ];
+        use std::ptr::null;
+        struct SyncWrapper<T>(T);
+        unsafe impl<T> Sync for SyncWrapper<T> {}
```

Comment:
Shouldn't this have a `#[repr(C)]` or something like that for good measure?

Reply:
The code here uses `.0.as_ptr()` to get a pointer to the array wrapped in the `SyncWrapper`. So the `repr` of the `SyncWrapper` type shouldn't matter.

Reply:
Oh right, that's good then. :+1: